site/pages/concepts/middleware.mdx

```diff
@@ -160,6 +160,61 @@ app.frame('/foo', (c) => {/* ... */})
 app.frame('/foo/bar', (c) => {/* ... */})
 ```
 
+### Example: TransactionResponse
+
+You can also write a custom middleware to add additional params to Frog's TransactionResponse (mimics [`Transaction Specs API`](https://warpcast.notion.site/Frame-Transactions-Public-9d9f9f4f527249519a41bd8d16165f73#8235c261091e4499b417e71dd1753e05)).
+
+```tsx twoslash
+// @noErrors
+import { Frog } from 'frog'
+ 
+export const app = new Frog()
+
+app.transaction( // [!code focus]
+  '/mint', // [!code focus]
+  async (c, next) => { // [!code focus]
+    await next(); // [!code focus]
+    const txParams = await c.res.json(); // [!code focus]
+    txParams.attribution = false; // [!code focus]
+    c.res = new Response(JSON.stringify(txParams), { // [!code focus]
+      headers: { // [!code focus]
+        'Content-Type': 'application/json', // [!code focus]
+      }, // [!code focus]
+    }); // [!code focus]
+  }, // [!code focus]
+  (c) => {/* ... */} // [!code focus]
+); // [!code focus]
+```
+
+### Example: Modifying the FrameResponse
+
+You can write a custom middleware to modify the response to add extra [open-frames metadata](https://github.com/open-frames/standard) to fit your needs.
+
+> Since the Frames Spec is still new (launched on 1/26/24) and hasn't hit a stable v1 yet (current vNext), we don't want to introduce additional standards to Frog that might conflict with or diverge from what the Farcaster team and community has planned. In addition, the vast majority of Frame usage at the moment is via Warpcast (and Farcaster) so we want to make sure Frog serves that main audience first.
+
+```tsx twoslash
+// @noErrors
+import { Frog } from 'frog'
+ 
+export const app = new Frog()
+
+app.use(async (c, next) => { // [!code focus]
+  await next(); // [!code focus]
+  const isFrame = c.res.headers.get('content-type')?.includes('html'); // [!code focus]
```

Comment:
If there is a better way to check if response is a frame, please update this 🙏

Reply:
Maybe extracting headers and checking if `fc:` or `frog:` are present?

Reply:
True, but it would also require an additional conditional statement to check if `content-type` is not `application/json` in case of transaction responses.

Reply:
true, but for extending meta tags responses maybe it would make sense to add such to `FrameResponse` itself?

for such middlewares lots of checks have to be performed carefully which is not really dx friendly.